File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = defaultdict(int)
        top_k = []
        for num in nums:
            freq[num] +=1

        freq =  dict(sorted(freq.items(), key=lambda item: item[1], reverse=True))

        for i in range(k):
            top_k.append(list(freq)[i])
        
        print(f"{top_k}")

        return top_k

File: Data Structures & Algorithms/special-array-i/submission-0.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int s = 0;
        int e = 1;
        
        if(nums.size() == 1){
            return true;
        }

        while(e < nums.size()){
            if( (nums[s] & 1) ==  (nums[e] & 1) ){
                return false;
            }
            s+=1;
            e+=1;
        }
        return true;

    }
};